docs/content/preview/develop/best-practices-ysql.md

```diff
@@ -87,6 +87,62 @@ Use [table partitioning](../../explore/ysql-language-features/advanced-features/
 For more details, see [Partition data by time](../common-patterns/timeseries/partitioning-by-time/).
 {{</tip>}}
 
+## Use multi row inserts wherever possible
+If you're inserting multiple rows, it's faster to batch them together whenever possible. You can start with 128 rows per-batch
+and test different batch sizes to find the sweet spot.
+
+Don't use multiple statements:
+```postgresql
+INSERT INTO users(name,surname) VALUES ('bill', 'jane');
+INSERT INTO users(name,surname) VALUES ('billy', 'bob');
+INSERT INTO users(name,surname) VALUES ('joey', 'does');
+```
+Do group values into a single statement:
+```postgresql
+INSERT INTO users(name,surname) VALUES ('bill', 'jane'), ('billy', 'bob'), ('joe', 'does');
+``` 
+
+
+## UPSERT multiple rows wherever possible
+
+PostgreSQL and YSQL enable you to do upserts using the `INSERT ON CONFLICT` clause. Similar to multi-row inserts, 
+you can also batch multiple upserts in a single `INSERT ON CONFLICT` statement for better performance.
+
+In case the row already exists, you can access the existing values using `EXCLUDED.<column_name>` in the query.
+
+In the example below we have a table with where we are keeping counters of different products and incrementing rows in batches:
+
+```postgresql
+CREATE TABLE products
+  (
+     name     TEXT PRIMARY KEY,
+     quantity BIGINT DEFAULT 0
+  );  
+---
+INSERT INTO products(name, quantity) 
+VALUES 
+  ('apples', 1), 
+  ('oranges', 5) ON CONFLICT(name) DO UPDATE 
+SET 
+  quantity = products.quantity + excluded.quantity;
+---
+INSERT INTO products(name, quantity) 
+VALUES 
+  ('apples', 1), 
+  ('oranges', 5) ON CONFLICT(name) DO UPDATE 
+SET 
+  quantity = products.quantity + excluded.quantity;
+---
+SELECT *
+FROM   products;
+  name   | quantity
+---------+----------
+ oranges |   	10
```

Comment:
spacing. is there a tab here?

Reply:
Fixed in https://github.com/yugabyte/yugabyte-db/pull/19236/commits/50f6caa2e8c9d12833f42b6f33471e89eefe0220